autobump-core/src/main/java/com/github/autobump/core/model/usecases/BumpResolverUseCase.java

```diff
@@ -16,23 +18,47 @@
     private Set<Dependency> dependencies;
     private IgnoreRepository ignoreRepository;
     private VersionRepository versionRepository;
+    private SettingsRepository settingsRepository;
 
-    public Set<Bump> doResolve(){
-        return groupBumps(makeBumpSet());
+    public Set<Bump> doResolve(String repoName) {
+        return groupBumps(makeBumpSet(repoName));
     }
 
-    public Set<Bump> makeBumpSet() {
+    public Set<Bump> makeBumpSet(String repoName) {
         Set<Bump> bumps = new HashSet<>();
         for (Dependency dependency : dependencies) {
             Version latestVersion = getUpdateVersion(dependency);
-            if (latestVersion != null && dependency.getVersion().compareTo(latestVersion) > 0) {
-                //newer version is found => make bump
+            if (latestVersion != null
+                    && dependency.getVersion().compareTo(latestVersion) > 0
+                    && !isIgnored(dependency, latestVersion, repoName)) {
+                //newer version is found && bump is not ignored => make bump
                 bumps.add(new Bump(Set.of(dependency), latestVersion));
             }
         }
         return bumps;
     }
 
+    @SuppressWarnings("checkstyle:ReturnCount")
+    private boolean isIgnored(Dependency dependency, Version latestVersion, String repoName) {
+        var settings = settingsRepository.getAllIgnores().stream()
+                .filter(setting -> setting.getRepositoryName().equalsIgnoreCase(repoName))
+                .collect(Collectors.toList());
+        for (Setting setting : settings) {
+            String[] dep = setting.getKey().split(":");
+            if (dep.length == 2) {
```

Comment:
The dep.length informs whether the ignore relates to a minor or a major ignore, right? Perhaps better to use the values in the settings to check this?